### PR TITLE
fix warnings

### PR DESCRIPTION
**Pull request by jackschofield23**
### Description

Fixes # .

Fixes a lot of the warnings currently in the projct

### Status
**Work in progress**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### CODE REVIEW

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 1]
```diff
@@ -596,7 +596,7 @@ private async Task<WorkflowInstance> CreateWorkflowInstanceAsync(WorkflowRequest
                 OutputDirectory = $"{payloadId}/workflows/{workflowInstanceId}/{executionId}",
                 ResultMetadata = { },
                 InputParameters = newInputParameters,
-                PreviousTaskId = previousTaskId
+                PreviousTaskId = previousTaskId is not null ? previousTaskId : string.Empty
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:599]: Could this be simplified to...

`previousTaskId ?? string.Empty`

?

**Review by neildsouth — APPROVED** [review]

File: src/WorkflowManager/Controllers/WorkflowInstanceController.cs [hunk 2]
```diff
@@ -23,110 +23,112 @@
...
+                if (workflowInstance is null)
+                {
+                    _logger.LogDebug($"{nameof(GetByIdAsync)} - Failed to find workflow instance with Id: {id}");
 
-            return Ok(workflowInstance);
-        }
-        catch (Exception e)
-        {
-            return Problem($"Unexpected error occured: {e.Message}", $"/workflowinstances/{nameof(id)}", (int)HttpStatusCode.InternalServerError);
+                    return NotFound($"Faild to find workflow instance with Id: {id}");
```
Comment (lillie-dae) [inline src/WorkflowManager/Controllers/WorkflowInstanceController.cs:123]: "Faild"

Should be "Failed"

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowTaskArtefactStepDefinitions.cs [hunk 3]
```diff
@@ -56,7 +56,7 @@ public async Task GivenIHaveABucketInMinIO(string name)
         public async Task GivenIHaveAPayloadInTheBucket(string folderName, string bucketName, string payloadId)
         {
             _outputHelper.WriteLine("Retrieving pathname");
-            var pathname = Path.Combine(GetDirectory(), "DICOMs", folderName, "dcm");
+            var pathname = Path.Combine(GetDirectory() ?? "", "DICOMs", folderName, "dcm");
```
Comment (lillie-dae) [inline tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowTaskArtefactStepDefinitions.cs:59]: "" in this file should probably be string.Empty;

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Support/Assertions.cs [hunk 4]
```diff
@@ -99,13 +100,13 @@ public void AssertWorkflowIstanceMatchesExpectedTaskStatusUpdate(WorkflowInstanc
...
 
-        public void AssertPagination<T>(int count, string queries, T Response)
+        public static void AssertPagination<T>(int count, string queries, T? Response)
         {
-            var data = Response.GetType().GetProperty("Data").GetValue(Response, null) as ICollection;
-            var totalPages = Response.GetType().GetProperty("TotalPages").GetValue(Response, null);
-            var pageSize = Response.GetType().GetProperty("PageSize").GetValue(Response, null);
-            var totalRecords = Response.GetType().GetProperty("TotalRecords").GetValue(Response, null);
-            var pageNumber = Response.GetType().GetProperty("PageNumber").GetValue(Response, null);
+            var data = Response?.GetType()?.GetProperty("Data")?.GetValue(Response, null) as ICollection;
```
Comment (lillie-dae) [inline tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Support/Assertions.cs:105]: assigning `Response?.GetType()?` to a variable could simplify this code a bit.

**Review by lillie-dae — APPROVED** [review]

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs (hunk 1 above)
Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs]: Fixed

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowTaskArtefactStepDefinitions.cs (hunk 3 above)
Comment (jackschofield23) [reply tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowTaskArtefactStepDefinitions.cs]: Files been removed

File: src/WorkflowManager/Controllers/WorkflowInstanceController.cs (hunk 2 above)
Comment (jackschofield23) [reply src/WorkflowManager/Controllers/WorkflowInstanceController.cs]: Fixed

File: tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Support/Assertions.cs (hunk 4 above)
Comment (jackschofield23) [reply tests/IntegrationTests/WorkflowExecutor.IntegrationTests/Support/Assertions.cs]: Changed

**Pull request merged**